src/proxy/h2.rs

```diff
@@ -106,36 +126,16 @@ impl H2StreamWriteHalf {
     }
 }
 
-impl Drop for H2StreamReadHalf {
-    fn drop(&mut self) {
-        let mut half_dropped = Arc::new(());
-        std::mem::swap(&mut self.half_dropped, &mut half_dropped);
-        if Arc::into_inner(half_dropped).is_none() {
-            // other half already dropped
-            let left = self.active_count.fetch_sub(1, Ordering::SeqCst);
-            trace!(
-                "dropping H2StreamReadHalf, has {} active streams left",
-                left - 1
-            );
-        } else {
-            trace!("dropping H2StreamReadHalf, write half remains");
-        }
-    }
-}
-
-impl Drop for H2StreamWriteHalf {
+impl Drop for DropCounter {
     fn drop(&mut self) {
         let mut half_dropped = Arc::new(());
         std::mem::swap(&mut self.half_dropped, &mut half_dropped);
         if Arc::into_inner(half_dropped).is_none() {
```

Comment:
I know you aren't changing this but it seems like this logic isn't correct... If we get `none` back it means we are **not** the last strong reference so the other half isn't dropped. If we get the inner value back we are the last strong reference.